phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCStreamIT.java

```diff
@@ -168,6 +172,42 @@ public void testOnlyOneStreamAllowed() throws Exception {
         assertStreamStatus(conn, tableName, streamName, CDCUtil.CdcStreamStatus.ENABLING);
     }
 
+    @Test
+    public void testStreamMetadataWhenTableIsDropped() throws SQLException {
+        Connection conn = newConnection();
+        MetaDataClient mdc = new MetaDataClient(conn.unwrap(PhoenixConnection.class));
+        String schemaName = "\"" + generateUniqueName().toLowerCase() + "\"";
+        String tableName = SchemaUtil.getTableName(schemaName, "\"" + generateUniqueName().toLowerCase() + "\"");
+        String unescapedFullTableName = SchemaUtil.getUnEscapedFullName(tableName);
+        String create_table_sql = "CREATE TABLE  " + tableName + " ( k INTEGER PRIMARY KEY," + " v1 INTEGER, v2 DATE)";
+        conn.createStatement().execute(create_table_sql);
+        String cdcName = "\"" + generateUniqueName().toLowerCase() + "\"";
+        String cdc_sql = "CREATE CDC " + cdcName + " ON " + tableName;
+        conn.createStatement().execute(cdc_sql);
+        TaskRegionObserver.SelfHealingTask task =
+                new TaskRegionObserver.SelfHealingTask(
+                        taskRegionEnvironment,
+                        QueryServicesOptions.DEFAULT_TASK_HANDLING_MAX_INTERVAL_MS);
+        task.run();
+        String partitionQuery = "SELECT * FROM SYSTEM.CDC_STREAM WHERE TABLE_NAME='" + unescapedFullTableName + "'";
```

Comment:
Shall we also have same query from SYSTEM.CDC_STREAM_STATUS?

Reply:
These checks do that:
`Assert.assertNotNull(mdc.getStreamNameIfCDCEnabled(unescapedFullTableName));`